src/gui/qgsmapcanvas.h

```diff
@@ -866,6 +869,13 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView, public QgsExpressionContex
      */
     bool allowInteraction( QgsMapCanvasInteractionBlocker::Interaction interaction ) const;
 
+    /**
+     * Compute the actual extent needed to fully display the \a extent rotated by \a rotation.
+     *
+     * \since QGIS 3.30
+     */
+    QgsRectangle computeExtentForRotation( const QgsRectangle &extent, double rotation ) const;
```

Comment:
This method could be static, right? QgsMapCanvasUtils might be a better fit for it.

Reply:
It could, but I'd have to add a QSize parameter (or height and width) since the computed extent relies on the canvas current aspect ratio.